invisible_cities/io/dst_io.py

```diff
@@ -29,25 +29,28 @@ def load_dsts(dst_list, group, node):
     dsts = [load_dst(filename, group, node) for filename in dst_list]
     return pd.concat(dsts, ignore_index=True)
 
-def _make_tabledef(column_types : pd.Series, str_col_length : int=32) -> dict:
+def _make_tabledef(df : pd.DataFrame, str_col_length : int=32) -> dict:
+    column_types = df.dtypes
     tabledef = {}
     for indx, colname in enumerate(column_types.index):
         coltype = column_types[colname].name
         if coltype == 'object':
+            if df[colname].str.len().max() > str_col_length:
+                warnings.warn(f'dataframe contains strings longer than allowed', UserWarning)
             tabledef[colname] = tb.StringCol(str_col_length, pos=indx)
         else:
             tabledef[colname] = tb.Col.from_type(coltype, pos=indx)
     return tabledef
```

Comment:
Actually I don't think `_make_tabledef` cares about the length of the object. I would revert this back to the original version and perform the check in `store_pandas_as_tables`. What do you think?

Reply:
Agree. I added a check in `store_pandas_as_tables ` directly. I also modified to cast numpy records array to the table type if it was an already existing one. However I feel like this 3x search for strings and type castings can be somehow better optimized...